src/layout/layout.js

```diff
@@ -129,17 +129,7 @@
 
   };
 
-  /**
-   * Provide local version of matchMedia. This is needed in order to support
-   * monkey-patching of matchMedia in the unit tests. Due to peculiarities in
-   * PhantomJS, it doesn't work to monkey patch window.matchMedia directly.
-   *
-   * @private
-   */
-  MaterialLayout.prototype.matchMedia_ = function(query) {
-    return window.matchMedia(query);
-  };
-
+  window["materialLayoutScreenWidthMediaQuery"] = window.matchMedia(window.MaterialLayout.prototype.Constant_.MAX_WIDTH);
```

Comment:
Use something like 
```suggestion
  /** @const @private */
  MaterialLayout.screenWidthMediaQuery_ = window.matchMedia(/** @type{string} */(MaterialLayout.prototype.Constant_.MAX_WIDTH));
```

Reply:
We can't put the query on the MaterialLayout instance because the init will be called first

Reply:
What about this:
```javascript
  /** @const @private */
  const MEDIA_QUERY = window.matchMedia('(max-width: 1024px)');
  MEDIA_QUERY.onchange = screenSizeHandler;

  /**
   * Class constructor for Layout MDL component.
   * Implements MDL component design pattern defined at:
   * https://github.com/jasonmayes/mdl-component-design-pattern
   *
   * @constructor
   * @param {HTMLElement} element The element that will be upgraded.
   */
  var MaterialLayout = function MaterialLayout(element) {
    this.element_ = element;
    // Initialize instance.
    this.init();
  };
  window['MaterialLayout'] = MaterialLayout;
```

Reply:
Make the query first, and does the query really need to be on the window?

Reply:
I was suggesting a (static) class property, not an instance property. Based on the previous comment I assumed it needs to be stubbed by tests, so a file-scoped `const` won't work.